### PR TITLE
🪶 [GRPO] PPO Lite: Scale rewards by Std of Batch

### LINKED ISSUE #3920: Does the trl library support the Lite PPO algorithm?
### Method description

In the following article：
_Part I: Tricks or Traps? A Deep Dive into RL for LLM Reasoning_(https://arxiv.org/pdf/2508.08221v1)
the author proposed a simple algorithm called Lite PPO, which can surpass DAPO and GRPO. Can it be supported in the trl library? thanks

### Open source status

- [ ] The method implementation is available
- [ ] The model weights are available
- [ ] The training datasets are available

### Provide useful links for the implementation

_No response_

### PR DESCRIPTION
**Pull request by pramodith**
# What does this PR do?

Updates the `scale_rewards` parameter to accept three different valid values `none`, `group` and `batch` to determine the level at which the std of rewards will be aggregated.

With this feature users can now train a model using the PPOLite technique outlined in https://arxiv.org/html/2508.08221v1, by setting `scale_rewards` = `group` and `loss_type='bnpo'` (for token level loss aggregration).

Fixes #3920

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [X] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [X] Did you make sure to update the documentation with your changes?
- [X] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by qgallouedec — APPROVED** [review]
very very nice!

File: docs/source/paper_index.md [hunk 1]
```diff
@@ -60,3 +60,38 @@ trainer = SFTTrainer(
...
+2. aggregating loss over the total number of tokens
+
+can unlock the learning capability of critic-free policies using vanilla PPO loss. Their results demonstrate that this simple combination consistently improves performance, surpassing strategies like GRPO and [DAPO](https://huggingface.co/papers/2503.14476).
+
+TRL supports using these learnings to train a GRPO model by:
+
+```python
+from trl import GRPOConfig
+
+config = GRPOConfig(
```
Comment (qgallouedec) [inline docs/source/paper_index.md]:
```suggestion
training_args = GRPOConfig(
```

**Comment by myron** [comment]
@pramodith thank you for adding!

in paper_index example, shouldn't the slale_rewards be 'batch' instead of 'group' ?

https://github.com/huggingface/trl/blob/main/docs/source/paper_index.md#part-i-tricks-or-traps-a-deep-dive-into-rl-for-llm-reasoning-lite-ppo

**Comment by pramodith** [comment]
@myron yes you're right! My bad, created a PR to fix that.
